jsonrpc/http.go

```diff
@@ -45,6 +47,14 @@ func (h *HTTP) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if req.Header.Get("Content-Encoding") == "gzip" {
+		gz, err := gzip.NewReader(req.Body)
+		if err != nil {
+			http.Error(writer, "invalid gzip", http.StatusBadRequest)
+			return
+		}
+		req.Body = gz
+	}
```

Comment:
I strongly think we should reconsider the support for gzip in request body. This can be the source of some bigger problems:
- Decompression bomb: Attacker can try sending a huge amount of data gzipped.
- It's hard for the outer layer to inspect, filter or cache the requests if necessary.

Reply:
cc @rodrigo-pino